Source/RunActivity/Viewer3D/RollingStock/MSTSLocomotiveViewer.cs

```diff
@@ -2835,9 +2810,29 @@ protected int PercentToIndex(float percent)
             {
                 try
                 {
-                    var val = ControlDiscrete.Values[0] <= ControlDiscrete.Values[ControlDiscrete.Values.Count - 1] ?
-                        ControlDiscrete.Values.Where(v => (float)v <= percent + 0.00001).Last() : ControlDiscrete.Values.Where(v => (float)v <= percent + 0.00001).First();
-                    index = ControlDiscrete.Values.IndexOf(val);
+                    // Binary search process to find the control value closest to percent
+                    List<double> vals = ControlDiscrete.Values;
+                    // Check if control values were defined in reverse, reverse them back for this calculation
+                    // This is less efficient, so creators should be encouraged to not do this
+                    bool reversed = ControlDiscrete.Values[0] > ControlDiscrete.Values[ControlDiscrete.Values.Count - 1];
+                    if (reversed)
+                        vals.Reverse();
```

Comment:
This is reversing ControlDiscrete.Values, is this intentional? The problem is that cloning the list for this function is not good, since that would trigger unnecessary garbage collection. I think the best thing would be to reverse the list just once, maybe adding a Reversed variable to CVCWithFrames

Reply:
I haven't noticed anything in my collection with reversed values but the previous (linear search) code supported it-still unsure this is needed at all. I'll look for a good place during initialization of the cab view control to see if it's sorted in the right direction.